langchain/embeddings/openai.ts

```diff
@@ -24,6 +24,8 @@ interface ModelParams {
 export class OpenAIEmbeddings extends Embeddings implements ModelParams {
   modelName = "text-embedding-ada-002";
 
+  numDimensions = 1536;
```

Comment:
is it possible to avoid having to pass this in? Most people won't want to think about this when initializing an embedding class

its also tied to the model that is passed in. by specifying both (esp if there are defaults) it makes it real easy to get out of date

Reply:
Well we need this to be set somewhere, as it is necessary to know this number in order to create any kind of ANN index. We can magically read the length of the vector of the first text that you try to add to the vector store and use that, but feels really strange to do that?

Reply:
thats what we do in python - it feels a bit strange but it makes it really easy for users not to have to think about embedding lengths

Reply:
other option would be to have a map of embedding model to embedding length. but then becomes a pain to update that map...

Reply:
Okay, a map would go out of date, I can do the magical one, but will throw an error in addTexts if user tries to change embeddings sizes halfway

Reply:
thats fine!

Reply:
Updated now